From c1320662b447fe70cd3eb8246ee24e0cb6a63782 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Wed, 5 Feb 2014 15:41:51 +0100
Subject: [PATCH 023/509] ARM: zynq: Move of_clk_init from clock driver

https://github.com/analogdevicesinc/linux.git xcomm_zynq
commit c58218ff54aab92aaea537cd921e87aa8fcb36a9

Move of_clk_init() from clock driver to enable
options not to use zynq clock driver.
Use for example fixed clock setting.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/mach-zynq/common.c |    2 ++
 drivers/clk/zynq/clkc.c     |    2 --
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-zynq/common.c b/arch/arm/mach-zynq/common.c
index 8e54319..74ce1d9 100644
--- a/arch/arm/mach-zynq/common.c
+++ b/arch/arm/mach-zynq/common.c
@@ -19,6 +19,7 @@
 #include <linux/cpumask.h>
 #include <linux/platform_device.h>
 #include <linux/clk.h>
+#include <linux/clk-provider.h>
 #include <linux/clk/zynq.h>
 #include <linux/clocksource.h>
 #include <linux/of_address.h>
@@ -132,6 +133,7 @@ static void __init zynq_timer_init(void)
 	zynq_early_slcr_init();
 
 	zynq_clock_init();
+	of_clk_init(NULL);
 	clocksource_of_init();
 }
 
diff --git a/drivers/clk/zynq/clkc.c b/drivers/clk/zynq/clkc.c
index c70a388..bfb7209 100644
--- a/drivers/clk/zynq/clkc.c
+++ b/drivers/clk/zynq/clkc.c
@@ -661,8 +661,6 @@ void __init zynq_clock_init(void)
 	of_node_put(slcr);
 	of_node_put(np);
 
-	of_clk_init(NULL);
-
 	return;
 
 np_err:
-- 
1.7.5.4

